### DISCUSSION TITLE
Why don't you support Selenium 4.

### DISCUSSION THREAD

**Discussion by skanda890** [microsoft/terminal#17051] [Q&A] [answered]
Why don't you support Selenium 4.

**Comment by DHowett** [comment, marked as answer]
We use WinAppDriver to run our UI-based tests. Selenium 4 uses the standardized W3C WebDriver protocol. WinAppDriver does not support the final standardized version of the W3C WebDriver protocol.

This is an issue that has been discussed at great, great length on the [WinAppDriver repository](https://github.com/microsoft/winappdriver).

https://github.com/microsoft/WinAppDriver/issues/1839

https://github.com/microsoft/WinAppDriver/issues/1550
